client/src/rpc_client.rs

```diff
@@ -161,7 +161,12 @@ impl RpcClient {
         signatures: &[Signature],
     ) -> RpcResult<Vec<Option<TransactionStatus>>> {
         let signatures: Vec<_> = signatures.iter().map(|s| s.to_string()).collect();
-        self.send(RpcRequest::GetSignatureStatuses, json!([signatures]))
+        self.send(
+            RpcRequest::GetSignatureStatuses,
+            json!([signatures, {
+                "searchTransactionHistory": true
```

Comment:
@mvines just wanted to run this change by you. Any objections to this for all client `getSignatureStatuses` queries?

Reply:
I have some (minor) concerns:
1. lack of symmetry with web3.js
2. we'll be hitting bigtable more on signature status checks
3. the web3.js version of deploy doesn't need this

Reply:
1 & 3. Yes, it's true that this code was already asymmetrical with web3.js, and still is. Worth asking if it should be. web3.js runs `sendAndConfirmTransaction` serially over all the program chunks; it doesn't need to search transaction history because the transaction is solidly in the status cache. However, when `solana deploy` is not throttled by the public api node rate limiter, it is noticeably faster than web3.js because it firehose-sends all the transactions and they can land in just a few slots. We could rewrite Cli to match the serial send-and-confirm -- and it would probably work pretty well against the public api nodes -- but I'm not excited about the slow-down for real use.

2. Fair... how big is your concern here? I could certainly add a `get_signature_statuses_with_history` method for `solana deploy`, so that other client `get_signature_statuses` uses don't hit bigtable. Or we could work a configuration into rpc to search blockstore only.
Or I could rewrite this entirely so that the confirmations happen in parallel with the sends.

Reply:
1. "`solana deploy` is faster", I'm sold :)
2. I think I favour a more conservative approach for now, `get_signature_statuses_with_history()`.  If it turns out that everybody wants `searchTransactionHistory: true` in general, we could just handle it on the server side and always assume `true`.

Reply:
✔️